### PR TITLE
Drop rust-Kathy

### PR DESCRIPTION
**Pull request by webbhorn**
We use typescript Kathy for everything now

### CODE REVIEW

**Comment by tkporter** [comment]
Could you remove it from all the infrastructure as well?

E.g. in the Helm chart:
* rm these values https://github.com/abacus-network/abacus-monorepo/blob/main/rust/helm/abacus-agent/values.yaml#L120
* any kathy files here https://github.com/abacus-network/abacus-monorepo/tree/main/rust/helm/abacus-agent/templates
* anything else kathy related in the helm chart

And then in `infra`:
* rm these configs that are no longer used https://github.com/abacus-network/abacus-monorepo/blob/main/typescript/infra/src/config/agent.ts#L142-L166, which should probably surface all usage of those types that can also be removed (i.e. also rm this which will show all places where it's referenced that can be removed https://github.com/abacus-network/abacus-monorepo/blob/main/typescript/infra/src/config/agent.ts#L214)

**Comment by webbhorn** [comment]
Yes, thanks for the pointers, I'll reply again after making those changes

**Comment by webbhorn** [comment]
Abandoning this for now.

File: rust/helm/abacus-agent/values.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 image:
   repository: gcr.io/clabs-abacus/abacus-agent
   pullPolicy: Always
-  tag: "latest"
```
Comment (nambrot) [inline rust/helm/abacus-agent/values.yaml:4]: It might be nice to move these kind of changes outside this PR

**Review by nambrot — APPROVED** [review]

**Comment by asaj** [comment]
From @tkporter:
> I think that PR's mostly fine but we should keep KEY_ROLE_ENUM.Kathy

File: typescript/infra/src/config/agent.ts [hunk 2]
```diff
@@ -472,16 +446,12 @@ export class ChainAgentConfig<Chain extends ChainName> {
...
   }
 
-  get kathyConfig(): KathyConfig | undefined {
-    if (!this.agentConfig.kathy) {
-      return undefined;
-    }
-    return getChainOverriddenConfig(this.agentConfig.kathy, this.chainName);
-  }
-
   get kathyEnabled() {
```
Comment (tkporter) [inline typescript/infra/src/config/agent.ts:449]: likewise I think you can safely remove the rest of the kathy stuff in this class

**Review by tkporter — APPROVED** [review]
lgtm but you can also remove the bits I mentioned

**Pull request merged**